deploy/call-roulette-deploy.yaml

```diff
@@ -0,0 +1,127 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: call-roulette
+---
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+ name: call-roulette
+ namespace: call-roulette
+ labels:
+   app: call-roulette
+spec:
+ replicas: 3
+ template:
+   metadata:
+     labels:
+       app: call-roulette
+   spec:
+     containers:
+     - name: call-roulette
+       image: juandebravo/call-roulette:0.2
+       ports:
+        - name: default-port
+          containerPort: 8080
+       resources:
+         requests:
+           memory: 100Mi
+         limits:
+           memory: 200Mi
+---
+kind: Service
+apiVersion: v1
+metadata:
+ labels:
+   app: call-roulette
+ name: call-roulette
+ namespace: call-roulette
+spec:
+ type: LoadBalancer
+ ports:
+ - port: 80
+   targetPort: 8080
+ selector:
+   app: call-roulette
+---
+---
+# https://github.com/kubernetes/ingress/blob/master/examples/deployment/nginx/nginx-ingress-controller.yaml
+# https://github.com/kubernetes/minikube/blob/master/deploy/addons/ingress/ingress-rc.yaml
+apiVersion: v1
+kind: ReplicationController
+metadata:
+  name: nginx-ingress-controller
+  namespace: kube-system
+  labels:
+    k8s-app: nginx-ingress-lb
+    kubernetes.io/cluster-service: "true"
+spec:
+  replicas: 1
+  selector:
+    k8s-app: nginx-ingress-lb
+    kubernetes.io/cluster-service: "true"
+  template:
+    metadata:
```

Comment:
To be "Prometheus Ready" :) add the following to the template metadata:
```
      annotations:
        prometheus.io/port: '10254'
        prometheus.io/scrape: 'true'
```
and you'll also need to include a configmap for the ingress controller with `enable-vts-status: true`

```
apiVersion: v1
data:
  enable-vts-status: "true"
kind: ConfigMap
metadata:
  name: nginx-ingress-lb-conf
  namespace: kube-system
```

make sure to load the custom config by adding the following to `args`:
`--configmap=$(POD_NAMESPACE)/nginx-ingress-lb-conf`